src/systems/visual/entity.ts

```diff
@@ -36,24 +36,42 @@ abstract class Entity {
 }
 
 export class NodeEntity extends Entity {
+  private _nodeGraphic: konva.Shape;
+
   private _color: string;
 
+  private _key: string | undefined;
+  get key(): string | undefined {
+    return this._key;
+  }
+  set key(value: string | undefined) {
+    this._key = value;
+    let system = GraphEngine.instance.systems.get(
+      GraphVisualSystem.name
+    )! as GraphVisualSystem;
+    [this.graphic, this._nodeGraphic] = NodeEntity.createNode(this.position.x, this.position.y, this.color, value);
+    system.scene.markNeedsDrawing();
+  }
+
   get color(): string {
     return this._color;
   }
 
   set color(color: string) {
+    this.graphic = this.graphic as konva.Circle;
```

Comment:
this.graphic is always a konva.Group, besides there's no need for this type assertion here.
```suggestion
```